src/components/HighlightText/HighlightText.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import {Text, StyleProp, TextStyle} from 'react-native'
+import styled from 'styled-components/native'
+import type {ITheme} from '../../theme'
+import {findAll} from './utils'
+
+type CustomTextStyleProp = StyleProp<TextStyle> | Array<StyleProp<TextStyle>>
+
+export type Theme = {
+  theme?: ITheme
+}
+
+export type HighlightTextProps = {
+  /**
+   * Text to highlight
+   */
+  textToHighlight: string
+  /**
+   * Array of search words
+   */
+  searchWords: Array<string>
+  /**
+   * custom function to process each word and text to highlight
+   * default: undefined
+   */
+  sanitize?: (string: string) => string
+  /**
+   * Escape special characters
+   * default: false
+   */
+  autoEscape?: boolean
+  /**
+   * Styles applied to sentence
+   * default: undefined
+   */
+  textWrapperStyle?: CustomTextStyleProp
+  /**
+   * Styles applied to normal text
+   * default: undefined
+   */
+  normalTextStyle?: CustomTextStyleProp
+  /**
+   * Styles applied to highlight text
+   * default: undefined
+   */
+  highlightTextStyle?: CustomTextStyleProp
+}
+
+const HighlightText: React.FC<HighlightTextProps> = ({
+  textToHighlight,
+  searchWords,
+  sanitize,
+  autoEscape = false,
+  textWrapperStyle,
+  normalTextStyle,
+  highlightTextStyle,
+  ...props
+}) => {
+  const chunks = findAll({textToHighlight, searchWords, sanitize, autoEscape})
+  return (
+    <TextWrapper style={textWrapperStyle} testID="container" {...props}>
```

Comment:
can we make testID is prop instead hardcode? 
exp: testID=`${testID}-container`

Reply:
i think it's not necessary because the testID only using for jest test